database/tracer.go

```diff
@@ -99,4 +106,111 @@ func (t ethDBTracer) AssertPersistedGenesisBlock(genesis core.Genesis) {
 			"persisted_balance", persistedBalance.String(),
 		)
 	}
+}
+
+func (t EthDBTracer) AssertPostTxSimulationState(from common.Address, tx *types.Transaction) {
+	t.logger.Infow("Tracing if ETH DB is in a valid state after simulation...", "chaindata", t.chainDataDir)
+
+	chainDb, err := ethdb.NewLDBDatabase(t.chainDataDir, 0, 0)
+	if err != nil {
+		t.logger.Errorw("unable to open LDB db", "err", err)
+		return
+	}
+	defer chainDb.Close()
+
+	headHash := rawdb.ReadHeadBlockHash(chainDb)
+	headNumber := rawdb.ReadHeaderNumber(chainDb, headHash)
+	block := rawdb.ReadBlock(chainDb, headHash, *headNumber)
+
+	stateDB, err := state.New(block.Root(), state.NewDatabase(chainDb))
```

Comment:
did you check if there is anyway to validate beforehand whether the chainDB is free to use so we prevent un-expected panics?